src/plugins/analysis/qemu_exec/code/qemu_exec.py

```diff
@@ -216,34 +224,124 @@ def process_qemu_job(file_path: str, arch_suffix: str, root_path: Path, results_
         }
 
 
+def _valid_execution_in_results(results: dict):
+    return any(
+        _output_without_error_exists(results[arch][option])
+        for arch in results
+        if 'error' not in results[arch]
+        for option in results[arch]
+        if option not in ['strace', 'error']
+    )
+
+
+def _output_without_error_exists(docker_output: Dict[str, str]) -> bool:
+    try:
+        return (
+            docker_output['stdout'] != ''
+            and (docker_output['return_code'] == '0' or docker_output['stderr'] == '')
+        )
+    except KeyError:
+        return False
+
+
 def test_qemu_executability(file_path: str, arch_suffix: str, root_path: Path) -> dict:
```

Comment:
I'd rename that, so that no test collector (nose, pytest etc.) finds this in the future.
`qemu_is_executable`?